Update mobile UI screenshots and consolidate docs (#12269)
newrelic/docs-website: src/content/docs/mobile-monitoring/mobile-monitoring-ui/crashes/handled-exceptions-analyze-trends-prevent-crashes.mdx

---

> title: 'Handled exceptions: Analyze trends, prevent crashes, view specific occurances'

Author: ```suggestion
title: 'Handled exceptions: Analyze trends, prevent crashes, drill down into occurrences'
```

import mobileHandledExceptionLocations from 'images/mobile_screenshot-full_handled-exception-locations.png'

import mobileHandledExceptionsOccurrences from 'images/mobile_screenshot-full_handled-exception-occurrences.png'

import mobileHandledExceptionLocationExample from 'images/mobile_screenshot-full_handled-exception-location-example.png'

import mobileEventTrailHandledExceptions from 'images/mobile_screenshot-full_event-trail-handled-exceptions.png'

> Our **Handled exceptions** feature help you identify significant factors contributing to poor mobile application experience, and use filterable data to find a resolution more quickly. You can also use the [handled exceptions API](#hex-api) to customize the data you send, and use NRQL to query and share the data.

Reviewer: Should it be "helps you"?

> Our **Handled exceptions** feature help you identify significant factors contributing to poor mobile application experience, and use filterable data to find a resolution more quickly. You can also use the [handled exceptions API](#hex-api) to customize the data you send, and use NRQL to query and share the data.

Reviewer: I think "handled exceptions" should be lowercase. 
https://docs.newrelic.com/docs/style-guide/capitalization/product-capability-feature-usage/

Author: @paperclypse But it's the name of the UI... still think it should be lowercased?
<img width="605" alt="image" src="https://user-images.githubusercontent.com/42753584/228020988-ca3ce790-f331-4dea-8273-30847ebb1e15.png">

Reviewer: Yeah, it should still be lowercased unless you are specifically referring to the UI in a procedure, such as "click *Handled exceptions*, then such and such"

When just referring to it more broadly, it should be lowercase

Author: Ah I see. Okay I'll fix it in another PR.

> Our **Handled exceptions** feature help you identify significant factors contributing to poor mobile application experience, and use filterable data to find a resolution more quickly. You can also use the [handled exceptions API](#hex-api) to customize the data you send, and use NRQL to query and share the data.

Author: ```suggestion
Our **Handled exceptions** feature helps you identify significant factors contributing to poor mobile application experience, and use filterable data to find a resolution more quickly. You can also use the [handled exceptions API](#hex-api) to customize the data you send, and use NRQL to query and share the data.
```

>   **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Mobile > (select an app) > Exceptions > Handled exceptions:** As you explore the wealth of data in the charts and table, use groups and filters to discover patterns that help you determine the root cause of mobile app exceptions.

Reviewer: Nice caption text!

</figcaption>

## Features

Handling exceptions as they occur can help improve your mobile app users' experience, but it's not enough to catch exceptions. You also need to know how to prevent them. Our Handled exceptions feature helps you answer these kinds of questions:

> * How many different types of handled exceptions are occurring? A high occurrence rate may necessitate changes to the back-end systems.

Reviewer: It's "backend" https://docs.newrelic.com/docs/style-guide/word-choice/usage-dictionary/#backend

Also, maybe make that sentence more active? Something like "When you're seeing a lot of handled exceptions, you may need to make changes to your backend systems"

> * How many different types of handled exceptions are occurring? A high occurrence rate may necessitate changes to the back-end systems.

Author: ```suggestion
* How many different types of handled exceptions are occurring? A high occurrence rate may indicate you need to make changes to the back-end systems.
```

* Why does the user's app usage result in a try/catch? What is the context for the exceptions?
* When can a test environment's responses to handled exceptions indicate additional, more serious problems?
* What would have caused a crash if the exception had not been caught in production? What else (in the code or back-end API) is still affecting the users' experience?

By using handled exceptions, you can identify and resolve these kinds of issues more quickly.

## Requirements

To view handled exception data, your app needs to meet these requirements:

* **Android:** [Android agent version 5.15.0 or higher](/docs/release-notes/mobile-release-notes/android-release-notes)
* **iOS:** [iOS agent version 5.15.0 or higher](/docs/release-notes/mobile-release-notes/ios-release-notes)

> Note that how long your mobile data is available for depends on the data retention you've set for mobile data. 

Reviewer: I'm not sure how much "note that" adds to a sentence :)

> Note that how long your mobile data is available for depends on the data retention you've set for mobile data. 

Author: ```suggestion
```
